src/blocks/battery.rs

```diff
@@ -221,7 +221,12 @@ impl UpowerDevice {
     /// if D-Bus cannot connect to this device, or if the device is not a
     /// battery.
     pub fn from_device(device: &str) -> Result<Self> {
-        let device_name = if device == "DisplayDevice" { device.to_string() } else { format!("battery_{}", device) };
+        let device_name =
+            if device == "DisplayDevice" || !device.starts_with("BAT") {
```

Comment:
This is a good intermediate solution and I won't hold off merging with it in place, but it's possible that this will surface issues if batteries are not given `BAT*`-style names on all systems.

Reply:
I've changed the logic to either use ```DisplayDevice``` or otherwise search for ```device``` as suffix in ```upower --enumerate```